rules/rules_test.go

```diff
@@ -15,41 +15,63 @@ package rules
 
 import (
 	"fmt"
-	"math"
-	"path"
-	"regexp"
-	"strconv"
 	"strings"
 	"testing"
 	"time"
 
 	clientmodel "github.com/prometheus/client_golang/model"
 
-	"github.com/prometheus/prometheus/rules/ast"
-	"github.com/prometheus/prometheus/stats"
+	"github.com/prometheus/prometheus/promql"
 	"github.com/prometheus/prometheus/storage/local"
 	"github.com/prometheus/prometheus/storage/metric"
-	"github.com/prometheus/prometheus/utility/test"
 )
 
 var (
-	testEvalTime = testStartTime.Add(testSampleInterval * 10)
-	fixturesPath = "fixtures"
-
-	reSample  = regexp.MustCompile(`^(.*)(?: \=\>|:) (\-?\d+\.?\d*(?:e-?\d*)?|[+-]Inf|NaN) \@\[(\d+)\]$`)
-	minNormal = math.Float64frombits(0x0010000000000000) // The smallest positive normal value of type float64.
+	testSampleInterval = time.Duration(5) * time.Minute
+	testStartTime      = clientmodel.Timestamp(0)
 )
 
-const (
-	epsilon = 0.000001 // Relative error allowed for sample values.
-)
+func getTestValueStream(startVal clientmodel.SampleValue, endVal clientmodel.SampleValue, stepVal clientmodel.SampleValue, startTime clientmodel.Timestamp) (resultValues metric.Values) {
```

Comment:
Yes, while we are at it: http://blog.golang.org/package-names
I think the post has a good point regarding `util(s|ity)?` packages.

How about making a clearer separation between Prometheus's core components (storage, retrieval, promql, ...) and more generic packages, e.g.

```
retrieval/
storage/
promql/
pkg/
    httputil/
    strutil/
    testutil/
    log/
web/
```